packages/tql/src/index.ts

```diff
@@ -0,0 +1,84 @@
+import { createQueryBuilder, isTemplateStringsArray } from './utils';
+import { TqlError } from './error';
+import type { CompiledQuery, Init } from './types';
+import {
+  TqlIdentifiers,
+  TqlList,
+  TqlNode,
+  TqlParameter,
+  TqlQuery,
+  TqlFragment,
+  TqlTemplateString,
+  TqlValues,
+  type TqlNodeType,
+  TqlSet,
+} from './nodes';
+import { build } from './build';
+
+export type * from './nodes';
+export type * from './types';
+export { PostgresDialect } from './dialects/postgres';
+
+export const init: Init = ({ dialect: Dialect }) => {
+  return {
+    query: (strings, ...values): CompiledQuery => {
+      const query = parseTemplate(TqlQuery, strings, values);
+      const qb = createQueryBuilder();
+      const d = new Dialect(qb.appendToQuery, qb.appendToParams);
+      const preprocessed = d.preprocess(query);
+      build(d, preprocessed);
+      return d.postprocess(qb.query, qb.params);
+    },
+    fragment: (strings, ...values) =>
+      parseTemplate(TqlFragment, strings, values),
+    identifiers: (ids) => new TqlIdentifiers(ids),
```

Comment:
I originally had both `identifier` (accepted a single value) and `identifiers` (accepts multiple values), but I combined them into one -- mirroring the VALUES clause in case, where even if you only have one value it's still VALUES. In this case, it can be called both ways: 

```ts
identifiers(['hello', 'world']); // output when compiled to Postgres: "hello", "world"
identifiers('hello'); // output when compiled to Postgres: "hello"
```

I dunno, it seems weird to have two exports just to have the ability to call it with a single value vs. an array, and I'm not sure if making the single export singular or plural is more weird. Thoughts?

Reply:
I think I may be leaning toward `identifier` which accepts either a single value or array, though it's still kinda weird either way... <picture data-single-emoji=":thinkies:" title=":thinkies:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/thinkies/d164ba479fed9c0d.png" alt=":thinkies:" align="absmiddle"></picture>